cogent/src/Cogent/DataLayout/CodeGen.hs

```diff
@@ -415,12 +562,12 @@ genAlignedRangeSetter boxType bitRange = do
 Creates a C function to extract the contents of an
 AlignedBitRange from a Cogent boxed type.
      
-@alignedRangeGetter AlignedBitRange { bitSizeABR, bitOffsetABR, wordOffsetABR} functionNameIdentifier@
+@alignedRangeGetter boxType AlignedBitRange { bitSizeABR, bitOffsetABR, wordOffsetABR} functionNameIdentifier@
 should be the C function
 
 @
-static inline unsigned int get`functionNameIdentifier`(unsigned int *p) {
-  return (*(p + `wordOffsetABR`) >> `bitOffsetABR`) & `mask bitSizeABR`;
+static inline unsigned int get`functionNameIdentifier`(`boxType` p) {
+  return (p.data[`wordOffsetABR`] >> `bitOffsetABR`) & `mask bitSizeABR`;
```

Comment:
Just ooc, is there any particular reason why you prefer the array operation instead of plain pointer arithmetic?

Reply:
It doesn’t make a difference - when loaded into auto Corres, both go to pointer arithmetic. I think array syntax is easier to read.